truffle-config.js

```diff
@@ -18,10 +18,17 @@
  *
  */
 
-const HDWalletProvider = require('truffle-hdwallet-provider');
-const fs = require('fs');
-const mnemonic = fs.readFileSync(".secret").toString().trim();
-const infuraKey = fs.readFileSync(".infuraKey").toString().trim();
+const getProvider = (networkId) => {
+  const HDWalletProvider = require('truffle-hdwallet-provider');
+  const fs = require('fs');
+  const mnemonic = fs.readFileSync(`${__dirname}/.secret`).toString().trim();
+  const infuraKey = fs.readFileSync(`${__dirname}/.infuraKey`).toString().trim();
+  
+  if (networkId === 3) {
+    return new HDWalletProvider(mnemonic, `https://ropsten.infura.io/v3/${infuraKey}`);
+  }
+  return null;
```

Comment:
Or rather `throw` to have the error here instead of later "why is this thing null?".